### PR TITLE
Channel: display playlists for autogenerated channels + include channel tags in api

### PR DESCRIPTION
**Pull request by ChunkyProgrammer**
I was hoping to add channel tags to the api and noticed some issues with autogenerated channels 

/channel/UCpS0hJ21amjA5eBApEOiIFw/playlists?sort_by=albums_and_singles
image

/channel/UCpS0hJ21amjA5eBApEOiIFw/playlists?sort_by=created_playlists
image

/channel/UCXwcsxJcdxfvsFiVtrOgy4Q/playlists?sort_by=albums_and_singles
image

/channel/UCXwcsxJcdxfvsFiVtrOgy4Q/playlists?sort_by=created_playlists
image

Tags:
auto generated channel (Minecraft - Topic):
/api/v1/channels/UCQvWX73GQygcwXOTSf_VDVg?pretty=1&fields=tags

other channel (Mr Beast):
/api/v1/channels/UCX6OQ3DkcsbYNE6H8uQQuVA?pretty=1&fields=tags

### CODE REVIEW

File: src/invidious/routes/channels.cr [hunk 1]
```diff
@@ -94,15 +94,16 @@ module Invidious::Routes::Channels
...
-    sort_by = env.params.query["sort_by"]?.try &.downcase
-
     if channel.auto_generated
-      return env.redirect "/channel/#{channel.ucid}"
+      sort_options = {"albums_and_singles", "created_playlists"}
+      sort_by = env.params.query["sort_by"]?.try &.downcase || sort_options[0]
+    else
+      sort_options = {"last", "oldest", "newest"}
+      sort_by = env.params.query["sort_by"]?.try &.downcase || sort_options[0]
     end
```
Comment (SamantazFox) [inline src/invidious/routes/channels.cr:103]:
```suggestion
      sort_options = {"albums_and_singles", "created_playlists"}
    else
      sort_options = {"last", "oldest", "newest"}
    end

    sort_by = env.params.query["sort_by"]?.try &.downcase || sort_options[0]
```

File: src/invidious/channels/about.cr [hunk 2]
```diff
@@ -52,7 +54,12 @@ def get_about_info(ucid, locale) : AboutChannel
     banners = initdata["header"]["interactiveTabbedHeaderRenderer"]?.try &.["banner"]?.try &.["thumbnails"]?
     banner = banners.try &.[-1]?.try &.["url"].as_s?
 
-    description_node = initdata["header"]["interactiveTabbedHeaderRenderer"]["description"]
+    description_base_node = initdata["header"]["interactiveTabbedHeaderRenderer"]["description"]
+    # some channels have the description in a simpleText
+    # ex: https://www.youtube.com/channel/UCQvWX73GQygcwXOTSf_VDVg/
+    description_node = description_base_node.dig?("simpleText") || description_base_node
+
+    tags = initdata.dig?("header", "interactiveTabbedHeaderRenderer", "badges").try &.as_a.map(&.["metadataBadgeRenderer"]["label"].as_s) || [] of String
```
Comment (SamantazFox) [inline src/invidious/channels/about.cr:62]: Minor style thing ^^
```suggestion
    tags = initdata.dig?("header", "interactiveTabbedHeaderRenderer", "badges")
      .try &.as_a.map(&.["metadataBadgeRenderer"]["label"].as_s) || [] of String
```

**Review by SamantazFox — APPROVED** [review]
Looks good!

**Review by SamantazFox — APPROVED** [review]

**Comment by SamantazFox** [comment]
Deployed on the test instance!

Funny how that channel is restricted there:
image

**Comment by SamantazFox** [comment]
So, I've tried using the channel you gave as an example.

---

The home page still redirects to the list of videos by default, but the GUI doesn't show it (there is no "videos" text above "playlists):

image

---

Then, on the playlists page, the sort options are wrong 

image

**Comment by ChunkyProgrammer** [comment]
@SamantazFox issues should be fixed now.

**Pull request closed without merging**